src/jdk.httpserver/share/classes/sun/net/httpserver/LeftOverInputStream.java

```diff
@@ -99,19 +100,42 @@ public synchronized int read (byte[]b, int off, int len) throws IOException {
         return readImpl (b, off, len);
     }
 
+    @Override
+    public synchronized long skip(long n) throws IOException {
+        long remaining = n;
+        int nr;
+
+        if (n <= 0) {
+            return 0;
+        }
+
+        int size = (int)Math.min(MAX_SKIP_BUFFER_SIZE, remaining);
+        byte[] skipBuffer = new byte[size];
+        while (remaining > 0) {
+            nr = readImpl(skipBuffer, 0, (int)Math.min(size, remaining));
```

Comment:
I wonder if we should import lines 135-137 just before line 115:

```suggestion
            if (server.isFinishing()) {
                break;
            }
            nr = readImpl(skipBuffer, 0, (int)Math.min(size, remaining));
```

and then implement drain simply as:

```
    public boolean drain (long l) throws IOException {
        while (l > 0) {
            long skip = skip(l);
            if (skip <= 0) break; // might return 0 if isFinishing
            l -= skip;
        }
        return eof;
    }
```

Reply:
Yeah, it seems better that drain() would call the skip() that's been implemented.